src/Microsoft.ML.Data/Training/ITrainerEstimator.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.ML.Trainers
     /// <typeparam name="TTransformer">The type of the transformer returned by the estimator.</typeparam>
     /// <typeparam name="TModel">The type of the model parameters.</typeparam>
     public interface ITrainerEstimator<out TTransformer, out TModel> : IEstimator<TTransformer>
-        where TTransformer : ISingleFeaturePredictionTransformer<TModel>
```

Comment:
Also need to confirm if there is a better way than loosening this restriction here, since we don't want ISingleFeaturePredictionTransformer for MatrixFactorization

Reply:
A separate conversation is ongoing in terms of finding a better way to do this. It doesn't block merging this PR.
If further change ends up needing to be done, I can submit in a follow up PR.